php/WP_CLI/Bootstrap/AutoloaderStep.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace WP_CLI\Bootstrap;
+
+/**
+ * Abstract class AutoloaderStep.
+ *
+ * Abstract base class for steps that include an autoloader.
+ *
+ * @package WP_CLI\Bootstrap
+ */
+abstract class AutoloaderStep implements BootstrapStep {
+
+	/**
+	 * Store state for subclasses to have access.
+	 *
+	 * @var BootstrapState
+	 */
+	protected $state;
+
+	/**
+	 * Process this single bootstrapping step.
+	 *
+	 * @param BootstrapState $state Contextual state to pass into the step.
+	 *
+	 * @return BootstrapState Modified state to pass to the next step.
+	 */
+	public function process( BootstrapState $state ) {
+		$this->state = $state;
+
+		$found_autoloader = false;
+		$autoloader_paths = $this->get_autoloader_paths();
+
+		if ( false === $autoloader_paths ) {
+			// Skip this autoloading step.
+			return $state;
+		}
+
+		foreach ( $autoloader_paths as $autoloader_path ) {
+			if ( is_readable( $autoloader_path ) ) {
+				try {
+					require $autoloader_path;
+					$found_autoloader = true;
+				} catch ( \Exception $exception ) {
+					\WP_CLI::warning(
+						"Failed to load autoloader '{$autoloader_path}'. Reason: "
+						. $exception->getMessage(),
+						'bootstrap'
+					);
+				}
+			}
+		}
+
+		if ( ! $found_autoloader ) {
+			$this->handle_failure();
+		}
+
+		return $this->state;
+	}
+
+	/**
+	 * Check whether WP-CLI is being run from within a PHAR bundle.
+	 *
+	 * @return bool
+	 */
+	protected function is_inside_phar() {
```

Comment:
I wanted to make sure that the bootstrapping process does not depend on part of the bootstrapping process (i.e. loading utility functions) being completed already.

I am aware this is not DRY, but I wanted to err on the safe side, at least while we don't yet know how we might re-arrange the bootstrapping steps.